tests/unit/record_encoder/test_hybrid_record_encoder.py

```diff
@@ -0,0 +1,119 @@
+import tempfile
+from pathlib import Path
+from unittest.mock import patch
+
+import pytest
+from pinecone_text.sparse import BM25Encoder
+
+from canopy.knowledge_base.models import KBDocChunk
+from canopy.knowledge_base.record_encoder import HybridRecordEncoder, DenseRecordEncoder
+from canopy.models.data_models import Query
+from tests.unit.stubs.stub_dense_encoder import StubDenseEncoder
+
+
+@pytest.fixture(scope="module")
+def documents():
+    return [KBDocChunk(id=f"doc_1_{i}",
+                       text=f"Sample document {i}",
+                       document_id=f"doc_{i}",
+                       metadata={"test": i},
+                       source="doc_1")
+            for i in range(5)]
+
+
+@pytest.fixture(scope="module")
+def queries():
+    return [Query(text="Sample query 1"),
+            Query(text="Sample query 2"),
+            Query(text="Sample query 3")]
+
+
+@pytest.fixture(scope="module")
+def inner_dimension():
+    return 4
+
+
+@pytest.fixture(scope="module")
+def dense_record_encoder(inner_dimension):
+    return DenseRecordEncoder(StubDenseEncoder(dimension=inner_dimension))
+
+
+@pytest.fixture(scope="module")
+def bm_25_encoder_df_path(documents):
+    bm25 = BM25Encoder()
+    bm25.fit([doc.text for doc in documents])
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        tmp_path = str(Path(tmp_dir, "bm25_params.json"))
+        bm25.dump(tmp_path)
+        yield tmp_path
+
+
+@pytest.fixture(scope="module")
+def hybrid_encoder(dense_record_encoder, bm_25_encoder_df_path):
+    return HybridRecordEncoder(dense_record_encoder,
+                               bm_25_encoder_df_path=bm_25_encoder_df_path,
+                               batch_size=2)
+
+
+def test_dimension(hybrid_encoder, inner_dimension):
+    assert hybrid_encoder.dimension == inner_dimension
+
+
+def test_init_encoder_invalid_alpha(dense_record_encoder):
+    with pytest.raises(ValueError):
+        HybridRecordEncoder(dense_record_encoder, alpha=-1)
+    with pytest.raises(ValueError):
+        HybridRecordEncoder(dense_record_encoder, alpha=2)
+    with pytest.raises(ValueError):
+        HybridRecordEncoder(dense_record_encoder, alpha=0)
```

Comment:
```suggestion
    with pytest.raises(ValueError, match="sparse only):
        HybridRecordEncoder(dense_record_encoder, alpha=0)
```
See suggestion above

Reply:
Done